stackslib/src/chainstate/nakamoto/mod.rs

```diff
@@ -2514,6 +2519,78 @@ impl NakamotoChainState {
             tenure_height,
         )?;
 
+        if !block.is_first_mined() {
+            let parent_reward_cycle = pox_constants
+                .block_height_to_reward_cycle(
+                    burn_dbconn.context.first_block_height,
+                    parent_chain_tip
+                        .burn_header_height
+                        .try_into()
+                        .expect("Burn block height exceeded u32"),
+                )
+                .unwrap();
+            let my_reward_cycle = pox_constants
+                .block_height_to_reward_cycle(
+                    burn_dbconn.context.first_block_height,
+                    burn_header_height,
+                )
+                .unwrap();
+            if parent_reward_cycle != my_reward_cycle {
+                // execute `set-aggregate-public-key` using `clarity-tx`
+                let aggregate_public_key = clarity_tx
+                    .connection()
+                    .with_readonly_clarity_env(
+                        false,
+                        chain_id,
+                        ClarityVersion::Clarity2,
+                        StacksAddress::burn_address(false).into(),
+                        None,
+                        LimitedCostTracker::Free,
+                        |vm_env| {
+                            vm_env.execute_contract_allow_private(
+                                &boot_code_id(POX_4_NAME, false),
+                                "get-aggregate-public-key",
+                                &vec![SymbolicExpression::atom_value(Value::UInt(
+                                    parent_reward_cycle as u128,
+                                ))],
+                                true,
+                            )
+                        },
+                    )
+                    .ok()
+                    .map(|agg_key_value| {
+                        Value::buff_from(agg_key_value.expect_buff(33))
+                            .expect("failed to reconstruct buffer")
+                    })
+                    .expect("get-aggregate-public-key returned None");
+
+                clarity_tx
+                    .connection()
+                    .with_readonly_clarity_env(
+                        false,
+                        chain_id,
+                        ClarityVersion::Clarity2,
+                        StacksAddress::burn_address(false).into(),
+                        None,
+                        LimitedCostTracker::Free,
+                        |vm_env| {
+                            vm_env.execute_contract_allow_private(
+                                &boot_code_id(POX_4_NAME, false),
+                                "set-aggregate-public-key",
+                                &vec![
+                                    SymbolicExpression::atom_value(Value::UInt(
+                                        my_reward_cycle as u128,
```

Comment:
Might want to change this to u128::from(my_reward_cycle)

Reply:
Fixed